File: src/component-factorial/component-factorial.go
package main

import (
	_ "fmt"
	"os"
	"runtime"
	"strconv"
)

const (
	RANGE int = 1000
)

func factorial(n uint64) uint64 {
	if n > 1 {
		n = n * factorial(n-1)
	} else {
		return uint64(1)
	}
	return n
}

func main() {
	N, _ := strconv.Atoi(os.Args[1]) // Iterations
	iCPU := runtime.NumCPU()
	runtime.GOMAXPROCS(iCPU)

	for i := 0; i <= N; i++ {
		for j := 0; j < RANGE; j++ {
			go factorial(uint64(j))
		}
	}
}

File: src/component-factorial/component-factorial.c
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>

int factorial(int);

#define RANGE 1000

int main(int argc, char const *argv[])
{
	int i, j;
	const int N = atoi(argv[1]);
	int iCPU = omp_get_num_procs();
  omp_set_num_threads(iCPU);
	omp_set_dynamic(0);

	for (i = 0; i <= N; i++)
		for(j = 0; j < RANGE; j++)
			factorial(j);
	
	return 0;
}

int factorial(int num)
 {
   int fac = 1, n;
   #pragma omp parallel for reduction(*:fac)
   for(n = 2; n <= num; ++n)
     fac *= n;
   return fac;
 }